internal/server.go

```diff
@@ -0,0 +1,90 @@
+package killgrave
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+
+	"github.com/gorilla/mux"
+	"github.com/pkg/errors"
+)
+
+// Server definition of mock server
+type Server struct {
+	impostersPath string
+	router        *mux.Router
+}
+
+// NewServer initialize the mock server
+func NewServer(p string, r *mux.Router) *Server {
+	return &Server{
+		impostersPath: p,
+		router:        r,
+	}
+}
+
+// Run read all the files on the impostersPath and creates different
+// handlers for each imposter
+func (s *Server) Run() error {
+	if _, err := os.Stat(s.impostersPath); os.IsNotExist(err) {
+		return errors.Wrapf(err, "the directory %s doesn't exists", s.impostersPath)
+	}
+	var imposterFileCh = make(chan string)
+	var done = make(chan bool)
+
+	go func() {
+		findImposters(s.impostersPath, imposterFileCh)
+		done <- true
+	}()
+	for {
+		select {
+		case f := <-imposterFileCh:
+			var imposters []Imposter
+			err := s.unmarshalImposters(f, &imposters)
+			if err != nil {
+				log.Printf("error trying to load %s imposter: %v", f, err)
+			} else {
+				s.createImposterHandler(imposters, f)
+				log.Printf("imposter %s loaded\n", f)
+			}
+		case <-done:
+			close(imposterFileCh)
+			close(done)
+			return nil
+		}
+	}
+}
+
+func (s *Server) createImposterHandler(imposters []Imposter, imposterFilePath string) {
```

Comment:
```suggestion
func (s *Server) addImposterHandler(imposters []Imposter, imposterFilePath string) {
```

Reply:
As the method has no return value (the handler itself) and it "adds" the handler into the server "configuration", I would change the method name. The "add" verb might not be the most proper verb, but I would choose another one (than create), at least to clarify that point.